lyluatex.lua

```diff
@@ -216,6 +210,72 @@ local function splitext(str, ext)
 end
 
 
+--[[ =============== Functions that output LaTeX code ===================== ]]
+
+function latex.filename(printfilename, insert, input_file)
+    if printfilename and input_file then
+        if insert == 'fullpage' then
+            warn('`printfilename` ignored with `insert=fullpage`')
+        else
+            local filename = input_file:gsub("(.*/)(.*)", "\\lyFilename{%2}\\par")
+            tex.sprint(filename)
+        end
+    end
+end
+
+function latex.fullpagestyle(style, ppn)
+    local function texoutput(s) tex.print('\\includepdfset{pagecommand='..s..'}') end
+    if style == '' then
+        if ppn then
+            texoutput('\\thispagestyle{empty}')
+        else texoutput('')
+        end
+    else texoutput('\\thispagestyle{'..style..'}')
+    end
+end
+
+function latex.includepdf(pdfname)
+    tex.sprint('\\includepdf[pages=-]{'..pdfname..'}')
+end
+
+function latex.includesystems(filename, protrusion, indent, do_compile)
+    local f = io.open(filename..'.tex', 'r')
+    local texoutput = f:read('*a')
+    f:close()
+    f = io.open(filename..'.count', 'r')
+    local nsystems = tonumber(f:read('*a'))
+    f:close()
+    if nsystems == 1 and protrusion ~= 0 then
+        warn('Only one system, deactivating indentation.')
+        protrusion = protrusion - indent
+    end
+    if do_compile then  -- new compilation, update output-systems.tex
+        texoutput =
+            [[\ifx\preLilyPondExample\undefined\else\expandafter\preLilyPondExample\fi]]..
+            texoutput:gsub([[\includegraphics{]], string.format(
+                [[\noindent\hspace*{%spt}\includegraphics{%s]],
+                protrusion, dirname(filename)))..
+            [[\ifx\postLilyPondExample\undefined\else\expandafter\postLilyPondExample\fi]]
+        f = io.open(filename..'.tex', 'w')
+        f:write(texoutput)
+        f:close()
+    end
+    tex.sprint(texoutput:explode('\n'))
+end
+
+function latex.label(label, labelprefix)
+    if label then tex.sprint('\\label{'..labelprefix..label..'}%%') end
+end
+
+function latex.verbatim(verbatim, ly_code, intertext, version)
+    if verbatim then
+        ly.verbprint(ly_code:explode('\n'))
+        if version then tex.sprint('\\lyVersion{'..version..'}') end
```

Comment:
Is it uncontroversial that the version is printed *after* the verbatim code?

Reply:
I would have expected it before, as that is the convention with LilyPond.  Not that I think the program actually cares.